addons/im_livechat/models/mail_message.py

```diff
@@ -41,28 +41,23 @@ def _to_store(self, store: Store, **kwargs):
                     .sudo()
                     .search([("mail_message_id", "=", message.id)], limit=1)
                 )
-                if chatbot_message.script_step_id:
-                    store.add(
-                        message,
-                        {
-                            "chatbotStep": {
-                                "message": Store.one_id(message),
-                                "scriptStep": Store.one_id(chatbot_message.script_step_id),
-                                "chatbot": {
-                                    "script": Store.one_id(
-                                        chatbot_message.script_step_id.chatbot_script_id
-                                    ),
-                                    "thread": Store.one_id(channel),
-                                },
-                                "selectedAnswer": Store.one_id(
-                                    chatbot_message.user_script_answer_id
-                                ),
-                                "operatorFound": channel.chatbot_current_step_id.step_type
-                                == "forward_operator"
-                                and len(channel.channel_member_ids) > 2,
+                if step := chatbot_message.script_step_id:
+                    message_data = {
+                        "chatbotStep": {
+                            "message": Store.one_id(message),
+                            "scriptStep": Store.one_id(step),
+                            "chatbot": {
+                                "script": Store.one_id(step.chatbot_script_id),
+                                "thread": Store.one_id(channel),
                             },
+                            "isLast": step._is_last_step(channel),
+                            "operatorFound": step.step_type == "forward_operator"
+                            and len(channel.channel_member_ids) > 2,
```

Comment:
This can still become flat even if there is no actual `ChatbotStep` model.

`chatbotStep` key in message should just take the id

And then a separate store.add on `ChatbotStep`

Reply:
And once it is flat, one_id can be replaced by one